### DISCUSSION TITLE
How to stop the app from overriding other apps (Better Touch Tool)

### DISCUSSION THREAD

**Discussion by ELgit** [noah-nuebling/mac-mouse-fix#949] [Questions]
I did a support request through the website but didn't get any support even though I bought the app and a "milkshake".  I have the app set to capture nothing on the switches (buttons area completely blank). And yet it overrides out the mouse settings in BTT. 

If I disable Mac mouse fix, BTT works again, so mouse fix is doing something globally beyond scrolling that's conflicting with BTT. Does anyone know how to fix this? Thanks!

**Comment by kuyechao** [comment]
I'm concerned about this as I use logi option+... Interested in the same issue.

**Comment by noah-nuebling** [comment]
Hey there @ELgit, thanks for the feedback!

Sorry my support is very sporadic, and buying milkshakes has no effect on that. I'm sorry for the disappointment and for the late reply. I've refuned your purchase. I hope that's ok.

As for your request, MMF is not explicitly designed to work alongside other mouse drivers such as BetterTouchTool. And I haven't thoroughly tested how it works alongside such drivers.

However, the way that I intended Mac Mouse Fix to work is that when you delete all actions on the "Buttons" tab,  Mac Mouse Fix does not at all interfere with the processing of the signals that your mouse sends when you press a mouse button. So BetterTouchTool should work normally with your mouse buttons, if you delete all actions on the "Buttons" tab - if things are working as intended.

If things aren't working as intended, then this might be a bug, or some intricacy I'm currently not understanding.
I plan to look into this more at some point, but I'm not sure if or when I can fix it. As of now, I unfortunately don't have a good solution. 

Perhaps you can find out more in the following Issue Categories:
(They are collections of GitHub Issues, where people had similar problems to this one. I don't think there's a solution to your specific issue in there, but it's been a while since I looked at them, so you might try your luck.)

https://github.com/noah-nuebling/mac-mouse-fix/labels/Karabiner%20Coexistence https://github.com/noah-nuebling/mac-mouse-fix/labels/MOS%20Coexistence https://github.com/noah-nuebling/mac-mouse-fix/labels/LinearMouse%20Coexistence https://github.com/noah-nuebling/mac-mouse-fix/labels/Logitech%20Options%20Coexistence https://github.com/noah-nuebling/mac-mouse-fix/labels/Better%20Touch%20Tool%20Coexistence

I'd also be interested to hear which features exactly you'd like to use in BetterTouchTool. So maybe I could build similar features into Mac Mouse Fix. Thanks!

I hope it helps a little! Let me know if you have anymore feedback or if I might be able to help with anything else.

---

Here's what the  "Buttons" tab should look like after you deleted all actions, just to make totally sure we're on the same page:
<img width="700" alt="Screenshot 2024-05-15 at 23 04 57" src="https://github.com/noah-nuebling/mac-mouse-fix/assets/40808343/76a4f536-261c-4ee8-b9dd-c86478415226">

**Comment by ELgit** [comment]
Thanks Noah!

This GitHub issue is identical to mine:
https://github.com/noah-nuebling/mac-mouse-fix/issues/706

MMF is taking over wheel-tilt when it's mapped in BTT, maybe because MMF isn’t coded to exclude those 2 buttons? (Wheel tilt left and right), which BTT identifies as “scroll left” and “scroll right” 

I don’t know anything about scroll wheel tilt, but I did find this on coding to detect it. 
https://stackoverflow.com/questions/1083691/how-to-detect-mouse-wheel-tilt 

I found this, but no clue how it works:
https://www.highrez.co.uk/downloads/XMouseButtonControl.htm 

Eric

**Reply by noah-nuebling** [reply]
Ahh interesting, thank you very much Eric! As far as I understand, tilting the scrollwheel sideways doesn't look like a button press to macOS, instead it looks like you're scrolling on a horizontal scrollwheel. 

Mac Mouse Fix does let you 'uncapture' the scroll wheel such that other apps like BTT can handle scroll wheel events instead of Mac Mouse Fix. You can do this by configuring the "Scrolling" tab like this:

<img width="455" alt="Screenshot 2024-05-15 at 23 48 25" src="https://github.com/noah-nuebling/mac-mouse-fix/assets/40808343/664ec170-5632-48cb-b9c2-7b4f8121d833">

However, the problem is that MMF doesn't let you 'uncapture' the horizontal scroll wheel events (which are sent when you tilt the wheel) separately from the vertical scroll wheel events (which are sent when you normally scroll the wheel).

Possible solutions to this might be to make it so that MMF lets you configure the handling of horizontal scrollwheel events separately from the handling of vertical scrollwheel events. An even better solution would be to reprogram the mouse's onboard memory such that tilting the wheel sends button events to macOS instead of sending horizontal scrollwheel events. 

I don't have a timeframe for when or if I might be able to implement such features in MMF, but as far as I know, reprogramming the onboard memory might be possible with the apps SteerMouse, BetterMouse, or Logitech Options. (But such reprogramming is not possible on all mouse models.)

I hope this helps a little or clears things up! Let me know if you have more questions.

**Comment by ELgit** [comment]
Sounds like you’ve figured it out, and separate control of horizontal and vertical scroll could eliminate the override problem. No hurry on implementation. 

I agree about programmable onboard memory. My Elecom Huge would be soooooo much better if it had that. It’s a pain -  I have to run both steermouse and BTT to see all 10 buttons and remap them to do what I want. That said, I can do 10 separate operations on top of the trackball - that’s hard to beat. 

I’m using a separate Logitech slimblade for scrolling and spreading trackball work across left/right hands. FYI, scrolling by turning a big fat trackball is totally addictive - and your app is killer for it! 

Regarding programming scroll wheel tilt, I have tilt left and right mapped to different functions, depending on the app I’m using. It’s mapped to copy and paste, or paste with existing formatting, or duplicate a calendar event, or go to today in the calendar, or to reposition different app windows where I want them across my screens, etc. I do a lot with wheel tilt, but I can’t program it at the source onboard memory. I can only re-map the command it’s sending, and that’s horizontal scroll, per Mac OS.   

Eric

**Reply by noah-nuebling** [reply]
I'm glad you like scrolling with the trackball with MMF! I designed the Click and Drag features with trackballs in mind, but I never tested it, because I don't have a trackball, so I'm glad to hear it works well for you. Thanks for the info!

That does sound like a pretty complicated but interesting setup. I definitely wanna try a Trackball too at some point.

If I added app-specific settings in MMF and made it possible to remap scrollwheel tilt to custom actions, would that allow you to simplify the setup?

**Comment by Bigshowzyb** [comment]
i also have this problem, overrides out the mouse settings in BTT, Logic Master 3S

**Reply by noah-nuebling** [reply]
Hey there @Bigshowzyb, thanks for the feedback! Which buttons' settings are being overridden by MMF? Does it still happen after you [uncapture](https://github.com/noah-nuebling/mac-mouse-fix/discussions/112) the button in Mac Mouse Fix?

**Comment by ELgit** [comment]
Thanks for the reply. The Elecom Huge is often less than $50 used. It’s a good platform for development because it has more buttons than any trackball. It’s so cheap, I have a backup unit. You don’t need to add a bunch of functionality to MMF unless you want to. For my needs, Steermouse and BTT already have app-specific mapping, and between Steermouse and BTT, I can “see” and remap the 10 Elecom Huge buttons so I’m good there. The thing I need most in order to use MMF is to have MMF separate horizontal and vertical scroll (as you concluded) so it doesn’t override scroll wheel tilt in BTT. That’ll help the others having the same issue. People use BTT because it has a massive list of functions mappable per-app, but it doesn’t have per-device mappings (just generic mouse, trackpad and Apple mouse). People use Steermouse because it has per-device mapping (and app specific) but it doesn’t have BTT’s massive list of functions. You have the “killer function” neither of them have: Apple trackpad-grade smooth scrolling. EricOn May 16, 2024, at 7:01 AM, Noah Nuebling ***@***.***> wrote:﻿
I'm glad you like scrolling with the trackball with MMF! I designed the Click and Drag features with trackballs in mind, but I never tested it, because I don't have a trackball, so I'm glad to hear it works well for you. Thanks for the info!
That does sound like a pretty complicated but interesting setup. I definitely wanna try a Trackball too at some point.
If I added app-specific settings in MMF and made it possible to remap scrollwheel tilt to custom actions, would that allow you to simplify the setup?
